quick_start/kusto_sample_app.py

```diff
@@ -0,0 +1,261 @@
+from azure.kusto.data import KustoClient, KustoConnectionStringBuilder, ClientRequestProperties
+
+from azure.kusto.ingest import (
+    QueuedIngestClient,
+    IngestionProperties,
+    FileDescriptor,
+    BlobDescriptor,
+    StreamDescriptor,
+    DataFormat,
+    ReportLevel,
+    IngestionMappingType,
+    KustoStreamingIngestClient,
+)
+
+# Todo - Config (Auto-Filled properties):
+kustoUri = "https://yogiladadx.westeurope.dev.kusto.windows.net"
+ingestUri = "https://ingest-yogiladadx.westeurope.dev.kusto.windows.net"
+databaseName = "e2e"
+tableName = "SampleTable"
+tableSchema = "SampleTable_schema"
+# Todo - Learn More: For additional information about supported data formats, see
+#  https://docs.microsoft.com/en-us/azure/data-explorer/ingestion-supported-formats
+fileFormat = DataFormat.CSV
+
+# Todo - Config (Optional): Change the authentication method from Device Code (User Prompt) to one of the other options
+#  Some of the auth modes require additional environment variables to be set in order to work (check the use below)
+#  Managed Identity Authentication only works when running as an Azure service (webapp, function, etc.)
+authenticationMode = "deviceCode"  # choose between: (deviceCode|managedIdentity|AppKey|AppCertificate)
+
+
+def main():
+    print("kusto sample app is starting")
+
+    # Todo - Learn More: For additional information on how to authorize users and apps on Kusto Database see:
+    #  https://docs.microsoft.com/en-us/azure/data-explorer/manage-database-permissions
+    if authenticationMode == "deviceCode":
+        print("During the run you will be prompted by the browser to enter a device code.")
+        print("Return to the console of the sample app and copy the code to provide from it.")
+        input("Press enter to continue...")
+
+    # Todo - Tip: Avoid creating a new Kusto Client for each use.
+    #  Create the clients once and use them as long as possible.
+    kusto_connection_string = create_connection_string(kustoUri, authenticationMode)
+    ingest_connection_string = create_connection_string(ingestUri, authenticationMode)
+    kusto_client = KustoClient(kusto_connection_string)
+    ingest_client = QueuedIngestClient(ingest_connection_string)
+
+    # Todo - Learn More: For additional information on how to create tables see: https://docs.microsoft.com/en-us/azure/data-explorer/one-click-table
+    print("")
+    print(f"Creating table '{databaseName}.{tableName}' if needed:")
+    # Todo (Yochai) what's the table creation command
+    command = ".show version"
+    if not run_control_command(kusto_client, databaseName, command):
+        print("Failed to create or validate table exists.")
+        exit(-1)
+
+    # Todo - Learn More: For additional information on Kusto Query Language see: https://docs.microsoft.com/en-us/azure/data-explorer/write-queries
+    print("")
+    print(f"Initial row count for '{databaseName}.{tableName}' is:")

```

Comment:
why is this important?

Reply:
It's not. Just an excuse to demo a simple query...